### PR TITLE
Linkerd install e2e

### PR DESCRIPTION
**Pull request by ilackarms**
BOT NOTES: 
resolves https://github.com/solo-io/supergloo/issues/354

### CODE REVIEW

File: pkg/install/linkerd/install_opts.go [hunk 1]
```diff
@@ -58,3 +65,15 @@ func (o *installOpts) install(ctx context.Context, installer kubeinstall.Install
 
 	return nil
 }
+
+func linkerdInstallerCallbacks() kubeinstall.CallbackOptions {
+	return &kubeinstall.CallbackOption{
+		OnPostCreate: func(res *unstructured.Unstructured) error {
+			// need to add some additional wait for linkerd-prometheus deployment
+			if res.GetName() == "linkerd-prometheus" && res.GetKind() == "Deployment" {
```
Comment (EItanya) [inline pkg/install/linkerd/install_opts.go]: empty if block?

File: test/e2e/linkerd/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	sgutils "github.com/solo-io/supergloo/test/e2e/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const superglooNamespace = "supergloo-system"
+
+var _ = Describe("linkerd e2e", func() {
+	meshName := "my-linkerd"
+
+	It("it installs linkerd", func() {
```
Comment (EItanya) [inline test/e2e/linkerd/e2e_test.go]: uninstall?

Comment (ilackarms) [reply test/e2e/linkerd/e2e_test.go]: it'll require a cli command, i dont mind doing as part of this pr

**Comment by bakerthomas317** [comment]
Okay thank you very much

**Pull request merged**